qiskit_ibm_runtime/utils/embeddings.py

```diff
@@ -0,0 +1,484 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=too-many-return-statements
+
+"""
+Utility class to represent an embedding of a set of qubits in a two-dimensional plane.
+"""
+
+from typing import List, Tuple, Union
+
+from qiskit.providers.backend import BackendV2
+from qiskit.transpiler import CouplingMap
+
+
+class Embedding:
+    r"""
+    A class to represent an embedding or arrangement of a set of qubits in a two-dimensional plane.
+
+    Args:
+        coordinates: A list of coordinates in the form ``(row, column)`` that specify the qubits'
+            location on the 2D plane.
+        coupling_map: A coupling map specifying how the qubits in the embedding are connected.
+    """
+
+    def __init__(
+        self,
+        coordinates: List[Tuple[int, int]],
+        coupling_map: Union[List[Tuple[int, int]], CouplingMap],
+    ) -> None:
+        num_qubits = len(coordinates)
+        if any(q0 > num_qubits or q1 > num_qubits for (q0, q1) in coupling_map):
```

Comment:
I assume `for (q0, q1) in coupling_map)` works if coupling map is a list of tuple or a CouplingMap?

Reply:
Yes, that right, CouplingMap is an iterable. This is also tested where I do, e.g., `e = Embedding.from_backend(self.vigo)`